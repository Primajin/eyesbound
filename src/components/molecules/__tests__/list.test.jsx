import React from 'react';
import {create} from 'react-test-renderer';

import List from '../list.jsx';

describe('List', () => {
	it('renders correctly with list of zero', () => {
		const data = [];
		const path = 'path';
		const title = 'title';
		const component = create(<List data={data} title={title} path={path}/>);
		expect(component).toMatchSnapshot();
	});

	it('renders correctly with list of one', () => {
		const listData = {data: {title: 'List Data Title'}, uid: '1234-5678-90'};
		const listDataNode = {node: listData};
		const data = [listDataNode];
		const path = 'path';
		const title = 'title';
		const component = create(<List data={data} title={title} path={path}/>);
		expect(component).toMatchSnapshot();
	});

	it('renders correctly with list of more', () => {
		const listData1 = {data: {title: 'List Data Title 1'}, uid: '1234-5678-90'};
		const listData2 = {data: {title: 'List Data Title 2'}, uid: '9876-5432-10'};
		const listDataNode1 = {node: listData1};
		const listDataNode2 = {node: listData2};
		const data = [listDataNode1, listDataNode2];
		const path = 'path';
		const title = 'title';
		const component = create(<List data={data} title={title} path={path}/>);
		expect(component).toMatchSnapshot();
	});
});
