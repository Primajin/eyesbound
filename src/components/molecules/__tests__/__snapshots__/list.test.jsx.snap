// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders correctly with a list of more 1`] = `
Array [
  <h1>
    title
     (
    2
    )
  </h1>,
  <ul>
    <li>
      <a
        aria-label="link-to-path"
        href="/path/1234-5678-90"
      >
        List Data Title 1
      </a>
    </li>
    <li>
      <a
        aria-label="link-to-path"
        href="/path/9876-5432-10"
      >
        List Data Title 2
      </a>
    </li>
  </ul>,
]
`;

exports[`List renders correctly with a list of one 1`] = `
Array [
  <h1>
    title
     (
    1
    )
  </h1>,
  <ul>
    <li>
      <a
        aria-label="link-to-path"
        href="/path/1234-5678-90"
      >
        List Data Title
      </a>
    </li>
  </ul>,
]
`;

exports[`List renders correctly with a list of zero 1`] = `
Array [
  <h1>
    title
     (
    0
    )
  </h1>,
  <ul />,
]
`;
